sentinel-ai-configured-agents/src/main/java/configuredagents/ConfiguredAgentFactory.java

```diff
@@ -5,31 +5,38 @@
 import com.phonepe.sentinelai.core.tools.ToolBox;
 import com.phonepe.sentinelai.toolbox.mcp.MCPToolBox;
 import com.phonepe.sentinelai.toolbox.remotehttp.HttpToolBox;
-import configuredagents.capabilities.*;
-import configuredagents.capabilities.impl.AgentMCPCapability;
-import configuredagents.capabilities.impl.AgentMemoryCapability;
-import configuredagents.capabilities.impl.AgentRemoteHttpCallCapability;
-import configuredagents.capabilities.impl.AgentSessionManagementCapability;
+import configuredagents.capabilities.AgentCapability;
+import configuredagents.capabilities.AgentCapabilityVisitor;
+import configuredagents.capabilities.impl.*;
 import lombok.Builder;
 import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
 
 import java.util.*;
 
 /**
  * Factory for creating instances of {@link ConfiguredAgent} based on the provided configuration
  */
+@Slf4j
 public class ConfiguredAgentFactory {
     private final SimpleCache<HttpToolBox> httpToolboxFactory;
     private final SimpleCache<MCPToolBox> mcpToolboxFactory;
+    private final CustomToolBox customToolBox;
 
     @Builder
     public ConfiguredAgentFactory(
-            @NonNull final HttpToolboxFactory httpToolboxFactory,
-            @NonNull final MCPToolBoxFactory mcpToolboxFactory) {
-        this.httpToolboxFactory = new SimpleCache<>(upstream -> httpToolboxFactory.create(upstream)
-                .orElseThrow(() -> new IllegalArgumentException("No HTTP tool box found for upstream: " + upstream)));
-        this.mcpToolboxFactory = new SimpleCache<>(upstream -> mcpToolboxFactory.create(upstream)
-                .orElseThrow(() -> new IllegalArgumentException("No MCP tool box found for upstream: " + upstream)));
+            final HttpToolboxFactory httpToolboxFactory,
+            final MCPToolBoxFactory mcpToolboxFactory,
+            final CustomToolBox customToolBox) {
+        this.httpToolboxFactory = null != httpToolboxFactory
```

Comment:
What if the user only wants httpToolBox and not mcp or the otherway around

Reply:
this is the implemented behaviour already. The formatting is weird leading to the confusion.